pkg/agent/reconcile_failover.go

```diff
@@ -66,58 +66,66 @@ func (f *failoverReconciler) RunForResource(ctx context.Context, req *ReconcileR
 		return nil
 	}
 
-	for _, va := range req.Attachments {
-		vaNode := va.Spec.NodeName
+	for _, peer := range req.Resource.Connections {
+		f.reconcilePeer(ctx, req, peer, recorder)
+	}
 
-		if vaNode == f.opt.NodeName {
-			continue
-		}
+	return nil
+}
 
-		connection := ""
-		for _, peer := range req.Resource.Connections {
-			if peer.Name == vaNode {
-				connection = peer.ConnectionState
-			}
-		}
+func (f *failoverReconciler) reconcilePeer(ctx context.Context, req *ReconcileRequest, peer DrbdConnection, recorder events.EventRecorder) {
```

Comment:
very minor nitpicking: a "peer" is not a "connection". also the `.Name` could be arbitrary. but we have that assumption that the name matches the peer so so many projects, we really should remove that from utils. just /me ranting about how bad the world is :)